language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/ExBudgeting.hs

```diff
@@ -0,0 +1,100 @@
+{-# LANGUAGE ConstraintKinds   #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE RankNTypes        #-}
+{-# LANGUAGE TemplateHaskell   #-}
+{-# LANGUAGE TypeApplications  #-}
+
+{- Note [Budgeting]
+
+When running Plutus code on the chain, you're running code on other peoples machines, so you'll have to pay for it. And it has to be possible to determine how much money that should be before sending the transaction with the Plutus code to the chain. If you spend too little, your transaction will be rejected. If you spend too much, you're wasting money. So it must be possible to estimate how much a script will cost. The easiest way to do so is to run the script locally and determine the cost. The functional nature of Plutus allows for the assumption it will cost a similar amount locally as on the chain. See 'CekBudgetMode'.
+
+Additionally, it's helpful to know which parts of the script cost how much. We assume it's useful to have a list of costs per term executed, so it's possible to estimate which parts of the script cost how much. The 'ExTally' has not been determined to be useful, but it was easy to implement.
+
+We're tracking execution cost via both memory (via 'ExMemory') and CPU (via 'ExCPU'). Node operators are more interested in space limits than time limits - the memory upper limit will be reached faster than the time limit (which would be until next block). The two resources are then converted to the main currency of the chain based on protocol parameters - that way it's possible to adjust the actual fees without changing the code.
```

Comment:
Not sure if this is true about them caring more about space vs time - I think they probably care about both.

Reply:
The closest thing we have to a policy is https://github.com/input-output-hk/plutus/blob/master/.editorconfig#L13, but I'm pretty sure we don't stick to that. I do think that we should avoid *excessively* long lines in code files, though. Most editors have support for reflowing comments with hard line breaks.

Reply:
vscode doesn't :-( I'll fix that.

Reply:
Fixed.